azurerm/data_source_arm_container_registry.go

```diff
@@ -0,0 +1,87 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmContainerRegistryRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		MigrateState:  resourceAzureRMContainerRegistryMigrateState,
+		SchemaVersion: 2,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMContainerRegistryName,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
```

Comment:
can we swap `resourceGroupNameSchema` for `resourceGroupNameForDataSourceSchema` here?